PolyEngine/Core/Src/SafePtrRoot.cpp

```diff
@@ -0,0 +1,46 @@
+#include "CorePCH.hpp"
+
+#include "SafePtrRoot.hpp"
+
+using namespace Poly;
+
+//TODO remove later
+int SafePtrRoot::Debug()
+{
+	return 5;
+}
+
+//------------------------------------------------------------------------------
+std::unordered_map<SafePtrRoot*, size_t> SafePtrRoot::PointersMap;
+Dynarray<SafePtrRoot*> SafePtrRoot::Pointers;
+
+//------------------------------------------------------------------------------
+SafePtrRoot::~SafePtrRoot()
+{
+	SafePtrRoot::ClearPointer(this);
+}
+
+//------------------------------------------------------------------------------
+size_t SafePtrRoot::RegisterPointer(SafePtrRoot *pointer)
+{
+	SafePtrRoot::Pointers.PushBack(pointer);
+	size_t idx = SafePtrRoot::Pointers.GetSize() - 1;
+	SafePtrRoot::PointersMap[pointer] = idx;
+	//return &SafePtrRoot::Pointers[idx];
+	return idx;
+}
+
+//------------------------------------------------------------------------------
+void SafePtrRoot::ClearPointer(SafePtrRoot *pointer)
+{
+	size_t idx = SafePtrRoot::PointersMap[pointer];
```

Comment:
Add assertion for lack of value in PointersMap. Use `getAt` instead of `operator[]`